File: Doodream/[1차]  캐시.js
function solution(cacheSize, cities) {
    var cache = [];
    var runTime = 0;
    cities = cities.map(x => x.toUpperCase())

    for (let i = 0; i < cities.length; i++) {
        var city = { name: cities[i], index: i };
        var isHit = cache.find(a => a.name === cities[i]);

        if (isHit) {
            var findIndex = cache.indexOf(isHit);
            cache[findIndex].index = i;
            runTime += 1;
        } else {

            cache.push(city);
            runTime += 5;
            if (cache.length > cacheSize) {
                cache.shift();
            }
        };

        cache.sort((a, b) => a.index - b.index);


    }

    var answer = runTime;
    return answer;
}

var cacheSize = 3;
var cities = ["Jeju", "Pangyo", "Seoul", "Jeju", "Pangyo", "Seoul", "Jeju", "Pangyo", "Seoul"]
console.log(solution(cacheSize, cities));

File: KwangHo/[1차] 캐시.py
def solution(cacheSize, cities):
    answer = 0
    arr=[]
    d=0
    for i in cities:
        i = i.upper()
        if i in arr :             #캐쉬히트
            d+=1
            arr.pop(arr.index(i)) #위치찾아서 팝
            arr.append(i)         #맨뒤에 추가
        else :                    #캐쉬미스   
            d+=5

            if cacheSize == 0 : continue    #캐쉬길이0예외처리
            if cacheSize == len(arr):       #옛날꺼 팝
                arr.pop(0)
            arr.append(i)                   #최근꺼 추가
            
    answer = d
    return answer